### PR TITLE
[wave2-842] hide empty related degrees when there are none.

### PR DESCRIPTION
**Pull request by mhughes2012**

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs_all_programs/views/defaults/ucgy_fgs_all_programs.view [hunk 1]
```diff
@@ -262,301 +262,3 @@ $handler->display->display_options['filters']['field_fgs_grad_degree_faculty_tar
 
 /* Display: Block */
 $handler = $view->new_display('block', 'Block', 'block');
-$handler->display->display_options['defaults']['group_by'] = FALSE;
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs_all_programs/views/defaults/ucgy_fgs_all_programs.view]: It seems that this commit is deleting the block view from Balraj's 868. Can this be restored? (e.g. pulling down latest develop should resolve this conflict).

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php [hunk 2]
```diff
@@ -190,12 +195,18 @@
                // link to the related degree program
                echo('<li><a href ="/node/' . $this_nid . '" >' . $this_cred_abbreviation . '<sup>' .  $current_grad_delivery_type_abbrev  . '</sup></a></li>');
 
+              // if we are here there is atleast one current related program. Pass it to JS
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: Need a space between words in "atleast" should read "at least"

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php [hunk 3]
```diff
@@ -246,7 +258,11 @@
                 // first letter needed for display
                 $this_grad_delivery_type_abbrev = substr($this_grad_delivery_type, 0, 1);
 
-                echo('<li><a href ="/node/' . $this_nid . '" >' . $this_cred_abbreviation . '<sup>' . $this_grad_delivery_type_abbrev . '</sup></a></li>');
+              // if we are here there is atleast one related program. Pass it to JS
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: Need a space between words in "atleast" should read "at least"

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php (hunk 2 above)
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: Also Pass it to JS should have a period to end the sentence.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php (hunk 3 above)
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: Also Pass it to JS should have a period to end the sentence.

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs_all_programs/views/defaults/ucgy_fgs_all_programs.view (hunk 1 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs_all_programs/views/defaults/ucgy_fgs_all_programs.view]: merged develop into this branch, no conflicts.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php (hunk 2 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: updated this comment accordingly.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php (hunk 3 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: updated this comment accordingly.

**Comment by dpfitzsi** [comment]
Fixed the text, looks like that view is still being deleted in this latest commit.

**Comment by dpfitzsi** [comment]
View is restored. Pinging Balraj again to look to verify.

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info [hunk 4]
```diff
@@ -24,9 +24,9 @@ dependencies[] = text
 dependencies[] = ucgy_programs
 dependencies[] = views_extra_pagers
 dependencies[] = view_unpublished
-stylesheets[all][] = css/ucgy_fgs_graduate_degree.css
 dependencies[] = views
 stylesheets[all][] = css/ucgy_fgs_graduate_degree.css
+scripts[] = js/ucgy_fgs_graduate_degree.js
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info]: Do we need this javascript js/ucgy_fgs_graduate_degree.js on all users and pages?

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php [hunk 5]
```diff
@@ -10,6 +10,11 @@
 // the credential abbreviation, e.g. "PhD"
 $grad_abbr = render($content['field_fgs_grad_degree_credential'][0]['#options']['entity']->field_credential_abbreviation['und'][0]['value']);
 
+// initialize current and related count for JS
+drupal_add_js(array('ucgy_fgs' => array('currentcount' => 0)), array('type' => 'setting'));
+drupal_add_js(array('ucgy_fgs' => array('relatedcount' => 0)), array('type' => 'setting'));
+
+
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: Remove this extra line.

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info (hunk 4 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info]: Thanks, @balrajb. I removed this from .info and and added a drupal_add_js to the node template.

**Comment by balrajb** [comment]
Conflicts need to resolve!

**Comment by mhughes2012** [comment]
merged through github.

**Comment by balrajb** [comment]
Conflicts resolved, approved!

**Comment by dpfitzsi** [comment]
Closed as this is no longer relevant

**Pull request closed without merging**